client/landing/gutenboarding/onboarding-block/designs/index.tsx

```diff
@@ -29,17 +29,27 @@ const Designs: React.FunctionComponent = () => {
 	const locale = useLocale();
 	const { goBack, goNext } = useStepNavigation();
 
-	const { setSelectedDesign, setFonts } = useDispatch( ONBOARD_STORE );
+	const { setSelectedDesign, setFonts, resetFonts } = useDispatch( ONBOARD_STORE );
 	const { getSelectedDesign, hasPaidDesign, getRandomizedDesigns } = useSelect( ( select ) =>
 		select( ONBOARD_STORE )
 	);
 	const isAnchorFmSignup = useIsAnchorFm();
 
+	const selectedDesign = getSelectedDesign();
+
 	useTrackStep( 'DesignSelection', () => ( {
-		selected_design: getSelectedDesign()?.slug,
+		selected_design: selectedDesign?.slug,
 		is_selected_design_premium: hasPaidDesign(),
 	} ) );
 
+	const [ userHasSelectedDesign, setUserHasSelectedDesign ] = React.useState( false );
+
+	React.useEffect( () => {
+		if ( selectedDesign && userHasSelectedDesign ) {
+			goNext();
+		}
+	}, [ goNext, userHasSelectedDesign, selectedDesign ] );
+
```

Comment:
This is to ensure that `goNext()` is not called prematurely when a design is selected.

Using `useEffect`, in combination with `userHasSelectedDesign`, makes sure that we use the updated version of `goNext`, which can correctly skip the `style` step if necessary.

Reply:
Done in https://github.com/Automattic/wp-calypso/pull/52166/commits/80869d2341fabf811d29bcad521eadce07379713